src/controller/CHIPDeviceController.cpp

```diff
@@ -321,10 +321,34 @@ CHIP_ERROR DeviceController::DisconnectDevice(NodeId nodeId)
 
 void DeviceController::OnFirstMessageDeliveryFailed(const SessionHandle & session)
```

Comment:
Wonder if this shouldn't be moved to `CASESessionManager` now that not only `DeviceController` manages devices. This code should be triggered even if two accessories (e.g. OTA Requestor to OTA Provider) lose the connection to each other.

Reply:
Probably makes sense, but as a separate PR, because if we do it naively now we will run into the issues described in https://github.com/project-chip/connectedhomeip/issues/16174

Filed https://github.com/project-chip/connectedhomeip/issues/16175 on this.